src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java

```diff
@@ -146,12 +151,16 @@ private void updateReadingGoal(int targetToRead, ReadingGoal.GoalType goalType)
 
         if (goalType.equals(ReadingGoal.GoalType.BOOKS)) {
             readingGoal.setText(haveRead + howManyReadThisYear + outOf + + targetToRead + getPluralized(" book", targetToRead));
-            goalProgress.setText(calculateProgress(targetToRead, howManyReadThisYear));
+            goalProgress.setText(calculateProgress(targetToRead, howManyReadThisYear, goalType));
             booksToRead.setText(calculateBooksToRead(targetToRead, howManyReadThisYear));
+            togglePagesGoalInfo(false, hasReachedGoal);
             toggleBooksGoalInfo(true, hasReachedGoal);    // show book goal-specific information
         } else {
             readingGoal.setText(haveRead + howManyReadThisYear + outOf + targetToRead + getPluralized(" page", targetToRead));
+            goalProgress.setText(calculateProgress(targetToRead, howManyReadThisYear, goalType));
```

Comment:
Since this line is a duplicate (it also appears in the `if` statement, this could be moved outside of the if/else